### PR TITLE
ThreadSafeCache Get method lock bug

### PR DESCRIPTION
**Pull request by blacktear23**
Make cache.Get use write lock because Get method will rearrange cache's entry.

### CODE REVIEW

File: server/cache/cache.go [hunk 1]
```diff
@@ -68,8 +68,8 @@ func (c *threadSafeCache) Put(key uint64, value interface{}) {
 
 // Get retrives an item from cache.
 func (c *threadSafeCache) Get(key uint64) (interface{}, bool) {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+	c.lock.Lock()
```
Comment (ngaut) [inline server/cache/cache.go]: Could you add a comment here? since Get method need a write lock is wired.

**Comment by siddontang** [comment]
Can this hurt the performance now? @disksing

**Comment by disksing** [comment]
It should be ok. It was `Lock` before https://github.com/pingcap/pd/pull/740.

**Comment by disksing** [comment]
CI failure not related to PD. I'm going to just merge it.

**Pull request merged**